packages/lesswrong/lib/collections/petrovDayActions/collection.ts

```diff
@@ -4,8 +4,30 @@ import { addUniversalFields, getDefaultResolvers } from '../../collectionUtils'
 import { getDefaultMutations, MutationOptions } from '../../vulcan-core/default_mutations';
 
 const options: MutationOptions<DbPetrovDayAction> = {
-  newCheck: (user: DbUser|null) => {
-    return true
+  newCheck: async (user: DbUser|null, document: DbPetrovDayAction|null) => {
+    const userRoleInfo = await PetrovDayActions.findOne({userId: user?._id, actionType: "hasRole"})
```

Comment:
needs an early return before this if there's no user or no document.  I'm not sure what a findOne with no userId does, but very plausibly "pull up a random record".